representation/transparency_using_selection/src/index.ts

```diff
@@ -13,20 +13,41 @@ async function init() {
 
     // Load PDB and create representation
     const trajectory = await plugin.builders.structure.parseTrajectory(fileData, "mmcif");
-    const data = await plugin.builders.structure.hierarchy.applyPreset(trajectory, "default");
+    const presetStateObjects = await plugin.builders.structure.hierarchy.applyPreset(trajectory, "default");
 
-    // Get Structure and StructureRef objects from data
-    const struct = data?.structure.cell?.obj?.data!;
-    const structRef = plugin.managers.structure.hierarchy.current.structures.find((sr)=>sr.cell.obj?.data === struct)
+    if (!presetStateObjects) {
+        throw new Error("Structure not loaded");
+    }
+
+    // Get Structure object from the structure stateObject selector.
+    // The Structure object contains properties and accessors to the underlying molecular data such as chains, residues, atoms, etc.
+    const struct = presetStateObjects.structure.data!;
+
+    // Get the StructureRef object from the Structure object. The StructureRef object is the
+    // part of the state that represents the structure. We use it to access the structure's components
+    // generated from the default preset.
+    const structRef = plugin.managers.structure.hierarchy.findStructure(struct)!;
     
-    // Query all chains with ID A or B
-    const ctx = new QueryContext(struct)
-    const query = Queries.generators.chains({
-        chainTest: l => StructureProperties.chain.label_asym_id(l.element) == 'A' || StructureProperties.chain.label_asym_id(l.element)== 'B'
-    })
-    const res = query(ctx);
+    // Create a selection for chains A and B
+
+        // A query defines the logic to select elements based on a predicate.
+        // The predicate is executed from a generator which updates the current element (here, an atom)
+        // in the query context, at each iteration. The predicate returns true if the element should be selected.
+        const query = Queries.generators.chains({
+            chainTest: ctx => {
+                const chainName = StructureProperties.chain.label_asym_id(ctx.element);
+                return chainName === 'A' || chainName === 'B';
+            }
+        })
+
+        const ctx = new QueryContext(struct);
+        const selection = query(ctx);
+
     // Convert the StructureSelection into a Loci
-    // Set a transparency of 1 (invisible) on the components using the loci
-    setStructureTransparency(plugin, structRef?.components!, 1, async ()=> StructureSelection.toLociWithSourceUnits(res))
+    const loci = StructureSelection.toLociWithSourceUnits(selection);
+
+    const transparency = 1; // 0 is fully opaque, 1 is fully transparent
+    // The helper takes care of updating the plugin state for each component based on the loci
```

Comment:
It appears that transparency is applied to a locus on a component? Is that right? Maybe we can make the comment more clear?